packages/turf-invariant/index.ts

```diff
@@ -235,7 +235,7 @@ export function collectionOf(
  */
 export function getGeom<G extends Geometries | GeometryCollection>(
   geojson: Feature<G> | G
-): G {
+): G | null {
```

Comment:
Here's the start of needing to be able to support that a Geometry can be null, which is allowed under the geojson spec, but Turf didn't support previously.

```
export interface Feature<G extends GeometryObject, P = GeoJsonProperties> extends GeoJsonObject {
    type: "Feature";
    /**
     * The feature's geometry
     */
    geometry: G | null;
```

https://datatracker.ietf.org/doc/html/rfc7946#section-3.2
```
A Feature object has a member with the name "geometry".  The value
      of the geometry member SHALL be either a Geometry object as
      defined above or, in the case that the Feature is unlocated, a
      JSON null value.
```

Reply:
I think I built this PR on a slightly older version of the `geojson` type package.  Turf already had a dependency on it in the top-level package.json but specified no version using only `"@types/geojson": "*"` and from what I can tell it was using a slightly older version as a result.  Once I pinned it to the latest version `7946.0.8` it picked up some nice additions.

I think it could simplify the initial migration, and we aren't forced to handle null right away.  I'm going to try and create a new smaller PR.

Specifically here is the latest `geojson` Feature type:
```
export interface Feature<G extends Geometry | null = Geometry, P = GeoJsonProperties> extends GeoJsonObject {
    type: "Feature";
    geometry: G;
    id?: string | number | undefined;
    properties: P;
}
```

Notice `null` is now added with extends instead of defined under the geometry property.  In practice this seems to allow downstream generic functions that use this to override and extend with something simpler that doesn't have null.

For example, the helper function `feature()`:
```
export function feature<G extends GeometryObject = Geometry, P = Properties>(
  geom: G,
  properties?: P,
  options: { bbox?: BBox; id?: Id } = {}
): Feature<G, P> {
```
It returns Feature, but it appears we can continue giving the generic G a lesser constraint that doesn't include `null`, so the incoming geometry just has to be like a `GeometryObject` which is to be like one of the 7 supported types `GeometryObject = Point | MultiPoint | LineString | MultiLineString | Polygon | MultiPolygon | GeometryCollection`
```.